osu.Game/Overlays/Profile/Sections/Historical/UserHistoryGraph.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Graphics;
+using static osu.Game.Users.User;
+
+namespace osu.Game.Overlays.Profile.Sections.Historical
+{
+    public class UserHistoryGraph : UserGraph<DateTime, long>
+    {
+        private UserHistoryCount[] values;
+
+        public UserHistoryCount[] Values
+        {
+            get => values;
+            set
+            {
+                values = value;
+                updateValues(value);
+            }
+        }
+
+        private readonly string tooltipCounterName;
+
+        public UserHistoryGraph(string tooltipCounterName)
+        {
+            this.tooltipCounterName = tooltipCounterName;
+        }
+
+        private void updateValues(UserHistoryCount[] values)
+        {
+            if (values == null || !values.Any())
+            {
+                Graph.FadeOut(FADE_DURATION, Easing.Out);
+                Data = null;
+                return;
+            }
+
+            Data = values.Select(v => new KeyValuePair<DateTime, long>(v.Date, v.Count)).ToArray();
+
+            if (values.Length > 1)
+            {
+                Graph.DefaultValueCount = Data.Length;
+                Graph.Values = Data.Select(x => (float)x.Value);
+                Graph.FadeIn(FADE_DURATION, Easing.Out);
+            }
```

Comment:
This chunk of code is almost shared with `RankGraph`. The only difference is how `Values` is calculated, which can be solved with a `Func<TValue, float>`.